### PR TITLE
Cut v2.22.1

### PR DESCRIPTION
**Pull request by brancz**
This cherry-picks #8061 and #8104, and cuts the v2.22.1 release.

@roidelapluie @brian-brazil @juliusv @gouthamve @codesome @bwplotka @beorn7 @simonpasquier

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 2.22.1 / 2020-11-03
+
+* [BUGFIX] Fix potential "mmap: invalid argument" errors. #8061
```
Comment (brian-brazil) [inline CHANGELOG.md:3]: This is a bit vague, can we describe what logic was fixed?

Comment (brancz) [reply CHANGELOG.md]: Added something more elaborate, let me know if that's better.

**Comment by roidelapluie** [comment]
Do you want #8139? Since 2.19 size based retention is not working.

**Comment by brancz** [comment]
I'm not in a hurry. If you think it'll get in soon, I'm happy to wait.

**Comment by roidelapluie** [comment]
> I'm not in a hurry. If you think it'll get in soon, I'm happy to wait.

It should be in today. Thanks!

File: CHANGELOG.md (hunk 1 above)
Comment (brian-brazil) [reply CHANGELOG.md]: I think it'd be good to mention that this is about the head chunks.

**Comment by codesome** [comment]
#8139 is now merged

**Comment by roidelapluie** [comment]
I opened #8147 with the cherry-pick as it is not straightforward

**Review by matthiasr — APPROVED** [review]
🥺

**Comment by brancz** [comment]
@brian-brazil @roidelapluie @codesome you all still ok with moving forward with this? I added the new changelog item from https://github.com/prometheus/prometheus/pull/8147.

**Comment by brian-brazil** [comment]
The code looks okay, my changelog comment above hasn't been fully responded to.

**Review by codesome — APPROVED** [review]

**Comment by brancz** [comment]
@brian-brazil thanks for pointing that out, I missed your reply to the comment.

File: CHANGELOG.md (hunk 1 above)
Comment (brancz) [reply CHANGELOG.md]: Added. Let me know how that is.

**Comment by brian-brazil** [comment]
:+1:

**Review by roidelapluie — APPROVED** [review]

**Pull request merged**